common/network-common/src/main/java/org/apache/spark/network/corruption/Cause.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.corruption;
+
+/**
+ * The cause of shuffle data corruption.
+ *
+ * @since 3.2.0
```

Comment:
is this really a public API? AFAIK the entire network module is private

Reply:
It's not a public API but it's a part of the protocol between Spark app and external shuffle service.

I'm thinking that we might move this to `spark-network-shuffle` module instead since it's related to shuffle.

Reply:
I have moved it to `spark-network-shuffle` module and removed the since tag. cc @mridulm